addons/l10n_ro/models/res_partner.py

```diff
@@ -26,8 +26,5 @@ def _compute_company_registry(self):
         super()._compute_company_registry()
         for partner in self.filtered(lambda p: p.country_id.code == 'RO' and p.vat):
             vat_country, vat_number = self._split_vat(partner.vat)
-            if vat_country.isnumeric():
-                vat_country = 'ro'
-                vat_number = partner.vat
-            if vat_country == 'ro' and self.simple_vat_check(vat_country, vat_number):
+            if vat_country in ('RO', ''):
```

Comment:
Shouldn't you also check that the format is valid?

Reply:
If it is not valid, it is not going to save, but of course, it could be saved later with another country's one.